packages/i18n/src/messages/localizedConstants.tsx

```diff
@@ -1658,3 +1659,103 @@ export const getIndigenousCommunity = (
     indigenousCommunity,
     `Invalid indigenous community '${indigenousCommunity}'`,
   );
+
+type StatusLabelKey =
+  | "DRAFT"
+  | "RECEIVED"
+  | "UNDER_REVIEW"
+  | "PENDING_SKILLS"
+  | "ASSESSMENT"
+  | "DATE_PASSED"
+  | "SCREENED_OUT"
+  | "QUALIFIED";
+
+const statusLabels = new Map<StatusLabelKey, MessageDescriptor | null>([
+  ["DRAFT", null],
+  [
+    "RECEIVED",
+    defineMessage({
+      defaultMessage: "Application received",
+      id: "4TmwRU",
+      description: "Status for an application that has been submitted",
+    }),
+  ],
+  [
+    "UNDER_REVIEW",
+    defineMessage({
+      defaultMessage: "Under review",
+      id: "wK5+0z",
+      description: "Status for an application that is being reviewed",
+    }),
+  ],
+  [
+    "PENDING_SKILLS",
+    defineMessage({
+      defaultMessage: "Pending skills assessment",
+      id: "+HxUqd",
+      description: "Status for an application that ie having skills reviewed",
+    }),
+  ],
+  [
+    "ASSESSMENT",
+    defineMessage({
+      defaultMessage: "Assessment in progress",
+      id: "nm1YKH",
+      description:
+        "Status for an application that where applicant is being assessed",
+    }),
+  ],
+  [
+    "DATE_PASSED",
+    defineMessage({
+      defaultMessage: "Submission date has passed",
+      id: "4KRs8G",
+      description:
+        "Status for an application that where the recruitment has expired",
+    }),
+  ],
+  [
+    "SCREENED_OUT",
+    defineMessage({
+      defaultMessage: "Screened out",
+      id: "njJCTd",
+      description:
+        "Status for an application that has been screened out of eligibility",
+    }),
+  ],
+  [
+    "QUALIFIED",
+    defineMessage({
+      defaultMessage: "Qualified",
+      id: "UayO6H",
+      description:
+        "Status for an application where the applicant has qualified",
+    }),
+  ],
+]);
+
+const statusLabelMap = new Map<PoolCandidateStatus, StatusLabelKey>([
+  [PoolCandidateStatus.Draft, "DRAFT"],
+  [PoolCandidateStatus.NewApplication, "RECEIVED"],
+  [PoolCandidateStatus.ApplicationReview, "UNDER_REVIEW"],
+  [PoolCandidateStatus.ScreenedIn, "PENDING_SKILLS"],
+  [PoolCandidateStatus.UnderAssessment, "ASSESSMENT"],
+  [PoolCandidateStatus.DraftExpired, "DATE_PASSED"],
+  [PoolCandidateStatus.ScreenedOutApplication, "SCREENED_OUT"],
+  [PoolCandidateStatus.ScreenedOutAssessment, "SCREENED_OUT"],
+  [PoolCandidateStatus.QualifiedAvailable, "QUALIFIED"],
+  [PoolCandidateStatus.QualifiedUnavailable, "QUALIFIED"],
+  [PoolCandidateStatus.QualifiedWithdrew, "QUALIFIED"],
+  [PoolCandidateStatus.PlacedCasual, "QUALIFIED"],
+  [PoolCandidateStatus.PlacedTerm, "QUALIFIED"],
+  [PoolCandidateStatus.PlacedIndeterminate, "QUALIFIED"],
+  [PoolCandidateStatus.Expired, "QUALIFIED"],
+  [PoolCandidateStatus.Removed, "QUALIFIED"],
+]);
+
+export const getPoolCandidateStatusLabel = (
```

Comment:
Hmm, this works quite differently from the other `getFoo` functions in this file.  It accepts a Maybe, doesn't use `getOrThrowError`, and returns null instead of an error message.  Did you intend to have this one work differently?

Reply:
Yes, it was fully intentional since we map multiple statuses to a single label. I debated even putting it here because it works differently 🤔 

Open to suggestions to either normalize this to match the pattern or moving it out of this file completely.

Reply:
No, I think this is a good place for it.  Maybe add a comment in case anyone else is wondering why it's different from the others, too.

Reply:
Good idea, added some documentation in [document status label helper](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5798/commits/1a577ef04df4fed7dd6480bf40bfcb7b9a131fc1)